dapp/src/components/page/governance/ProposalForm.tsx

```diff
@@ -181,23 +224,44 @@ const ProposalForm: React.FC = () => {
 
       const directoryCid = await client.uploadDirectory(files);
 
-      setIsLoading(false);
-      alert(
-        `Proposal submitted successfully! CID: ${getIpfsBasicLink(directoryCid.toString())}`,
+      const { createProposal } = await import("@service/WriteContractService");
+
+      const res = await createProposal(
```

Comment:
I feel that there should be some validation there to prevent calling the smart contract if IPFS upload failed.

Also, please see my suggestion in the IPFS issue to sign a challenge transaction and validate it backend side to get the IPFS token.